File: mobile/lib/features/shared/domain/user_summary.dart
/// Public profile + reputation slice returned by the backend on cards and
/// detail screens (`toUserSummary` in `src/utils/user-summary.ts`).
class UserSummary {
  const UserSummary({
    required this.id,
    required this.fullName,
    this.avatarUrl,
    this.homeCity,
    this.ratingAvg = 0,
    this.ratingCount = 0,
    this.deliveredCount = 0,
  });

  final String id;
  final String fullName;
  final String? avatarUrl;
  final String? homeCity;
  final double ratingAvg;
  final int ratingCount;
  final int deliveredCount;

  static UserSummary? fromJson(Object? json) {
    if (json is! Map) return null;
    return UserSummary(
      id: json['id'].toString(),
      fullName: (json['full_name'] ?? '').toString(),
      avatarUrl: _str(json['avatar_url']),
      homeCity: _str(json['home_city']),
      ratingAvg: (json['rating_avg'] as num?)?.toDouble() ?? 0,
      ratingCount: (json['rating_count'] as num?)?.toInt() ?? 0,
      deliveredCount: (json['delivered_count'] as num?)?.toInt() ?? 0,
    );
  }
}

String? _str(Object? v) {
  final s = v?.toString().trim() ?? '';
  return s.isEmpty ? null : s;
}

File: src/utils/user-summary.ts
/**
 * The public-facing slice of a user shown on cards and detail screens:
 * name, avatar and reputation. Never includes contact details or the email.
 */
export interface UserSummary {
  id: string;
  full_name: string;
  avatar_url: string | null;
  home_city: string | null;
  rating_avg: number;
  rating_count: number;
  delivered_count: number;
}

export interface UserSummaryRow {
  id: string;
  full_name: string;
  avatar_url?: string | null;
  home_city?: string | null;
  rating_sum?: number | string | null;
  rating_count?: number | string | null;
  delivered_count?: number | string | null;
}

export function toUserSummary(row: UserSummaryRow): UserSummary {
  const count = Number(row.rating_count ?? 0);
  const sum = Number(row.rating_sum ?? 0);
  return {
    id: row.id,
    full_name: row.full_name,
    avatar_url: row.avatar_url ?? null,
    home_city: row.home_city ?? null,
    rating_avg: count > 0 ? Math.round((sum / count) * 10) / 10 : 0,
    rating_count: count,
    delivered_count: Number(row.delivered_count ?? 0),
  };
}

/** Columns to select from `users` when a summary is needed. */
export const USER_SUMMARY_COLUMNS = [
  'id',
  'full_name',
  'avatar_url',
  'home_city',
  'rating_sum',
  'rating_count',
  'delivered_count',
] as const;
